provider/mediaconvert/mediaconvert_test.go

```diff
@@ -0,0 +1,643 @@
+	p := &mcProvider{client: client}
+
+	_ = p.Healthcheck()
+
+	if !client.listJobsCalled {
+		t.Fatal("expected Healthcheck() to call ListJobs")
+	}
+}
+
+func Test_mcProvider_JobStatus(t *testing.T) {
+	tests := []struct {
+		name        string
+		destination string
+		mcJob       mediaconvert.Job
+		wantStatus  provider.JobStatus
+		wantErr     bool
+	}{
+		{
+			name:        "a job that has been queued returns the correct status",
+			destination: "s3://some/destination",
+			mcJob: mediaconvert.Job{
+				Status: mediaconvert.JobStatusSubmitted,
+			},
+			wantStatus: provider.JobStatus{
+				Status:       provider.StatusQueued,
+				ProviderName: Name,
+				Output: provider.JobOutput{
+					Destination: "s3://some/destination/jobID/",
+				},
+			},
+		},
+		{
+			name:        "a job that is currently transcoding returns the correct status",
+			destination: "s3://some/destination",
+			mcJob: mediaconvert.Job{
+				Status:             mediaconvert.JobStatusProgressing,
+				JobPercentComplete: aws.Int64(42),
+			},
+			wantStatus: provider.JobStatus{
+				Status:       provider.StatusStarted,
+				ProviderName: Name,
+				Progress:     42,
+				Output: provider.JobOutput{
+					Destination: "s3://some/destination/jobID/",
+				},
+			},
+		},
+		{
+			name:        "a job that has finished transcoding returns the correct status",
+			destination: "s3://some/destination",
+			mcJob: mediaconvert.Job{
+				Status: mediaconvert.JobStatusComplete,
+				OutputGroupDetails: []mediaconvert.OutputGroupDetail{{
+					OutputDetails: []mediaconvert.OutputDetail{
+						{
+							VideoDetails: &mediaconvert.VideoDetail{
+								HeightInPx: aws.Int64(2160),
+								WidthInPx:  aws.Int64(3840),
+							},
+						},
+						{
+							VideoDetails: &mediaconvert.VideoDetail{
+								HeightInPx: aws.Int64(1080),
+								WidthInPx:  aws.Int64(1920),
+							},
+						},
+					},
+				}},
+			},
+			wantStatus: provider.JobStatus{
+				Status:       provider.StatusFinished,
+				ProviderName: Name,
+				Progress:     100,
+				Output: provider.JobOutput{
+					Destination: "s3://some/destination/jobID/",
+					Files: []provider.OutputFile{
+						{Height: 2160, Width: 3840},
+						{Height: 1080, Width: 1920},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			client := &testMediaConvertClient{
+				t:                   t,
+				jobReturnedByGetJob: tt.mcJob,
+			}
+
+			p := &mcProvider{client: client, cfg: &config.MediaConvert{
+				Destination: tt.destination,
+			}}
+
+			status, err := p.JobStatus(&defaultJob)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("mcProvider.JobStatus() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if g, e := status, &tt.wantStatus; !reflect.DeepEqual(g, e) {
+				t.Fatalf("mcProvider.JobStatus(): wrong job request\nWant %+v\nGot %+v\nDiff %s", e,
+					g, cmp.Diff(e, g))
+			}
+		})
+	}
+}
+
+type testMediaConvertClient struct {
```

Comment:
Can you move the test client to a separate file?

Reply:
Sure thing

Reply:
done